pkg/client/services.go

```diff
@@ -77,7 +77,13 @@ func (c *services) Get(name string) (result *api.Service, err error) {
 // Create creates a new service.
 func (c *services) Create(svc *api.Service) (result *api.Service, err error) {
 	result = &api.Service{}
-	err = c.r.Post().Namespace(c.ns).Resource("services").Body(svc).Do().Into(result)
+	// v1beta3 does not allow POST without a namespace.
+	needNamespace := !api.PreV1Beta3(c.r.APIVersion())
```

Comment:
I'm saying that anyone who has an empty string either got it from client input (in which case the server will catch it) or it will be caught lower in the code base.  No one should be passing an empty namespace, ever.  I don't think we need to test in the client (I have a pull incoming which normalizes all our clients now that the server is normalized).

----- Original Message -----
> > @@ -77,7 +77,13 @@ func (c *services) Get(name string) (result
> > *api.Service, err error) {
> >  // Create creates a new service.
> >  func (c *services) Create(svc *api.Service) (result *api.Service, err
> >  error) {
> >  	result = &api.Service{}
> > -	err =
> > c.r.Post().Namespace(c.ns).Resource("services").Body(svc).Do().Into(result)
> > +	// v1beta3 does not allow POST without a namespace.
> > +	needNamespace := !api.PreV1Beta3(c.r.APIVersion())
> 
> If not here, then we will need such conditions everywhere at higher level
> code.
> ```
> If (apiVersion = v1beta3 && ns == "") {
>   ns = api.NamespaceDefault
> }
> servicesInterface = client.services(ns)
> ```
> Is that better?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5587/files#r26879365